### PR TITLE
React tutorial

### PR DESCRIPTION
**Pull request by MarcoDiaz2000**
In this exercise, I consolidated my knowledge about React, components, props, state, and hooks, by following a tutorial.

### CODE REVIEW

File: src/components/Header.jsx [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+const Header = (props) => {
+  const headerStyle = {
+    padding: '20px 0',
+    lineHeight: '1.5em',
+    color: '#aeadad',
+    textAlign: 'center',
+  };
+  return (
+    <header style={headerStyle} className={styles.header}>
+      {props.children}
```
Comment (unorjikingsley) [inline src/components/Header.jsx:12]: - [ ] You did a good job working on your Header.jsx 💯 However, please make sure that the {props.children} has the prop validation  kindly use the code below to fix it. 🤓 🔧 

Header.propTypes = {
  children: PropTypes.node.isRequired,
};

Comment (unorjikingsley) [inline src/components/Header.jsx:12]: - [ ] You did a good job working on your Header.jsx 💯 However, please make sure that the {props.children} has the prop validation  kindly use the code below to fix it. 🤓 🔧 

Header.propTypes = {
  children: PropTypes.node.isRequired,
};

File: src/components/InputTodo.jsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
+import { useState } from 'react';
+import '@/styles/app.css';
+import { FaPlusCircle } from "react-icons/fa"
+import { IconContext } from "react-icons"
+
+
+const InputTodo = ({ addTodoItem }) => {
```
Comment (unorjikingsley) [inline src/components/InputTodo.jsx:6]: - [ ] You did a good job working on your InputTodo.jsx 💯 However, please make sure that the {addTodoItem} has the prop validation  kindly fix it. 🤓 🔧

File: src/components/NavbarFirst.jsx [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+            Services <span>&#8595;</span>
+          </button>
+          {dropdown && (
+            <ul>
+              <li>Design</li>
+              <li>Development</li>
+            </ul>
+          )}
+        </li>
+      </ul>
```
Comment (unorjikingsley) [inline src/components/NavbarFirst.jsx]: - [ ] Please don't forget to remove old pieces of commented code, they make your code base look ugly and make it less readable.  🤓 🔧

File: src/components/ProtectedRoute.jsx [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import { Navigate, useLocation } from 'react-router-dom';
+import { useAuthContext } from '@/context/AuthContext';
+
+const ProtectedRoute = ({ children }) => {
```
Comment (unorjikingsley) [inline src/components/ProtectedRoute.jsx:4]: - [ ] You did a good job working on your ProtectedRoute.jsx 💯 However, please make sure that the {children} has the prop validation  kindly fix it. 🤓 🔧

File: src/components/TodoApp.jsx [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+      <div className="todos">
+        <Navbar />
+        <Modal />
+        <Form />
+        <Header />
+        <TodosLogic />
+      </div>
+    </div>
+  );
+};
```
Comment (unorjikingsley) [inline src/components/TodoApp.jsx]: - [ ] Great Job. Always remember to remove old pieces of code for readability. 🤓

File: src/components/TodoItem.jsx [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+
+  let viewMode = {};
+  let editMode = {};
+  if (editing) {
+    viewMode.display = 'none';
+  } else {
+    editMode.display = 'none';
+  }
+
+  const handleUpdatedDone = (event) => {
```
Comment (unorjikingsley) [inline src/components/TodoItem.jsx:33]: - [ ] You did a good job working on your TodoItem.jsx 💯 However, there are lots of place that needs the prop validation
- [ ] In line 10, you declared `UpdateInput`  but never used it. kindly fix that also. 🤓 🔧

File: src/context/AuthContext.jsx [hunk 7]
```diff
@@ -0,0 +1,25 @@
+import { useEffect, useState, useContext, createContext } from 'react';
+
+const AuthContext = createContext(null);
+export const AuthProvider = ({ children }) => {
+  const [user, setUser] = useState('');
+  function getUsername() {
```
Comment (unorjikingsley) [inline src/context/AuthContext.jsx:6]: - [ ] You did a good job working on your AuthContect.jsx. 💯 However, you declared a getUsername function that was never used. 🤓

File: src/index.css [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+
+  color-scheme: light dark;
+  color: rgba(255, 255, 255, 0.87);
+  background-color: #242424;
+
+  font-synthesis: none;
+  text-rendering: optimizeLegibility;
+  -webkit-font-smoothing: antialiased;
+  -moz-osx-font-smoothing: grayscale;
+  -webkit-text-size-adjust: 100%;
```
Comment (unorjikingsley) [inline src/index.css:14]: - [ ] Beware of using code that is not supported in all browser. This is important for users so they dont have bad experience or different experience from others using the app.

**Review by unorjikingsley — CHANGES_REQUESTED** [review]
# STATUS:  CHANGES REQUIRED  ♻️ 🚧 

Hi @MarcoDiaz2000 , 


While you have done a great job working on this milestone 💻 , there are some issues that you still need to work on to go to the next project. ♻️ 

You are almost there, keep up the good work! 🚀 



## Highlights
✅ Great use of Git flow. 💯 
✅ Your project is documented professionally. 📜 
✅ Great UI design. 💻  

## Required Changes ♻️
_Check the comments under the review._
 - [ ] Please pay attention to props validation, remove commented lines of code and also remove functions that is not in use so that your code is more clean and readable to all.

You can also consider:
- [ ]  Next time, and for future commits, I kindly suggest that you please make your PR more descriptive. Focus on listing the feature in that PR to enable other programmers to understand the journey of building the app.
:x: Bad Commit Messages :x:


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by MarcoDiaz2000** [comment]
Hi, @unorjikingsley 
Ok, I will make the requested changes

Thank you

> # STATUS: CHANGES REQUIRED ♻️ 🚧
> Hi @MarcoDiaz2000 ,
> 
> While you have done a great job working on this milestone 💻 , there are some issues that you still need to work on to go to the next project. ♻️
> 
> You are almost there, keep up the good work! 🚀
> 
>  [ 68747470733a2f2f6d65646961322e67697068792e636f6d2f6d656469612f336f68687769323549535843375a34744d732f67697068792d646f776e73697a65642d6d656469756d2e676966 ](https://camo.githubusercontent.com/1ffb80c92a4b41eda3c73a360964b1cb423c621dedcf0c344ce0f24f8092e3a8/68747470733a2f2f6d65646961322e67697068792e636f6d2f6d656469612f336f68687769323549535843375a34744d732f67697068792d646f776e73697a65642d6d656469756d2e676966) [ ](https://camo.githubusercontent.com/1ffb80c92a4b41eda3c73a360964b1cb423c621dedcf0c344ce0f24f8092e3a8/68747470733a2f2f6d65646961322e67697068792e636f6d2f6d656469612f336f68687769323549535843375a34744d732f67697068792d646f776e73697a65642d6d656469756d2e676966)
> 
> ## Highlights
> ✅ Great use of Git flow. 💯 ✅ Your project is documented professionally. 📜 ✅ Great UI design. 💻
> 
> ## Required Changes ♻️
> _Check the comments under the review._
> 
> * [ ]  Please pay attention to props validation, remove commented lines of code and also remove functions that is not in use so that your code is more clean and readable to all.
> 
> You can also consider:
> 
> * [ ]   Next time, and for future commits, I kindly suggest that you please make your PR more descriptive. Focus on listing the feature in that PR to enable other programmers to understand the journey of building the app.
>   ❌ Bad Commit Messages ❌
> 
> ## Optional suggestions
> _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._
> 
> Cheers and Happy coding!👏👏👏
> 
> Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
> 
> **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
> 
> _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**